packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain/DestinationNetworkBox.tsx

```diff
@@ -24,158 +19,207 @@ import {
     app: { selectedToken }
   } = useAppState()
   const [networks] = useNetworks()
   const { childChain, childChainProvider, isDepositMode } =
     useNetworksRelationship(networks)
+  const nativeCurrency = useNativeCurrency({ provider: childChainProvider })
   const { isArbitrumOne } = isNetwork(childChain.id)
+  const isCctpTransfer = useIsCctpTransfer()
+
+  const isBatchTransferSupported = useIsBatchTransferSupported()
+  const { isAmount2InputVisible } = useAmount2InputVisibility()
 
   const { erc20ChildBalances } = useBalances()
   const nativeCurrencyBalances = useNativeCurrencyBalances()
   const selectedTokenBalances = useSelectedTokenBalances()
 
-  const nativeCurrency = useNativeCurrency({ provider: childChainProvider })
+  const selectedTokenDestinationBalance = isDepositMode
+    ? selectedTokenBalances.childBalance
+    : selectedTokenBalances.parentBalance
 
-  const isCctpTransfer = useIsCctpTransfer()
+  const selectedTokenOrNativeCurrencyBalance = selectedToken
+    ? selectedTokenDestinationBalance
+    : nativeCurrencyBalances.destinationBalance
 
-  if (selectedToken) {
-    return (
-      <>
-        <TokenBalance
-          balance={
-            isDepositMode
-              ? selectedTokenBalances.childBalance
-              : selectedTokenBalances.parentBalance
+  return (
+    <div
+      className="rounded px-3 text-white [&>*+*]:border-t [&>*+*]:border-gray-600"
+      style={{ backgroundColor: '#00000050' }}
+    >
+      {isCctpTransfer && isDepositMode && (
+        <BalanceRow
+          parentErc20Address={
+            isArbitrumOne
+              ? CommonAddress.Ethereum.USDC
+              : CommonAddress.ArbitrumOne.USDC
           }
-          on={isDepositMode ? NetworkType.childChain : NetworkType.parentChain}
-          forToken={selectedToken}
-          tokenSymbolOverride={
-            // we need to send the proper, sanitized token-name to the component
-            selectedToken?.symbol
-              ? sanitizeTokenSymbol(selectedToken?.symbol, {
-                  chainId: networks.destinationChain.id,
-                  erc20L1Address: selectedToken?.address
+          balance={formatAmount(
+            (isArbitrumOne
+              ? erc20ChildBalances?.[CommonAddress.ArbitrumOne.USDC]
+              : erc20ChildBalances?.[CommonAddress.ArbitrumSepolia.USDC]) ??
+              constants.Zero,
+            {
+              decimals: selectedToken?.decimals
+            }
+          )}
+          symbolOverride="USDC"
+        />
+      )}
+      <BalanceRow
+        parentErc20Address={selectedToken?.address}
+        balance={
+          selectedTokenOrNativeCurrencyBalance
+            ? formatAmount(selectedTokenOrNativeCurrencyBalance, {
+                decimals: selectedToken
+                  ? selectedToken.decimals
+                  : nativeCurrency.decimals
+              })
+            : undefined
+        }
+        symbolOverride={isCctpTransfer ? 'USDC.e' : undefined}
```

Comment:
if it's CCTP, and it's withdrawal mode, the destination chain's (parent chain) symbol should be USDC, because the source chain (child chain) USDC would be the native USDC.